src/ray/raylet/scheduling_resources.cc

```diff
@@ -168,6 +168,13 @@ void ResourceSet::SubtractResourcesStrict(const ResourceSet &other) {
   for (const auto &resource_pair : other.GetResourceAmountMap()) {
     const std::string &resource_label = resource_pair.first;
     const FractionalResourceQuantity &resource_capacity = resource_pair.second;
+
+    // It is valid to request 0 resources from a non-existent resource,
+    // so if that happens, we skip all the checking logic and updating below.
+    if (resource_capacity == 0) {
```

Comment:
I think this is a different bug. I thought we maintain as an inariant that `ResourceSet` has no resources with capacity 0.